src/main/java/address/sync/CloudSimulator.java

```diff
@@ -35,37 +36,41 @@ public CloudSimulator(boolean shouldSimulateUnreliableNetwork) {
      * Gets the updated data subjected to simulated random behaviors such as random changes,
      * random delays and random failures. The data is originally obtained from a given file.
      * The data is possibly modified in each call to this method and is persisted onto the same file.
-     * When failure condition occurs, this returns an empty data set.
+     * When failure condition occurs, this returns an empty optional
+     *
+     * @return optional wrapping the (possibly corrupted) data, or empty if retrieving failed.
      */
-    public AddressBook getSimulatedCloudData(File cloudFile) {
+    public Optional<AddressBook> getSimulatedCloudData(File cloudFile) {
         System.out.println("Simulating cloud data retrieval...");
-        AddressBook modifiedData = new AddressBook();
         try {
-            AddressBook data = XmlHelper.getDataFromFile(cloudFile);
+            AddressBook modifiedData;
+            AddressBook data = XmlFileHelper.getDataFromFile(cloudFile);
             if (!this.simulateUnreliableNetwork) {
-                return data;
+                return Optional.of(data);
             }
 
             // no data could be retrieved
             if (RANDOM_GENERATOR.nextDouble() <= FAILURE_PROBABILITY) {
                 System.out.println("Cloud simulator: failure occurred! Could not retrieve data");
-                AddressBook wrapper = new AddressBook();
-                wrapper.setPersons(new ArrayList<>());
-                wrapper.setGroups(new ArrayList<>());
-                return wrapper;
+                return Optional.empty();
             }
 
             modifiedData = simulateDataModification(data);
             modifiedData.getPersons().addAll(simulateDataAddition());
-            XmlHelper.saveDataToFile(cloudFile, modifiedData.getPersons(), modifiedData.getGroups());
+
+            XmlFileHelper.saveDataToFile(cloudFile, modifiedData.getPersons(), modifiedData.getGroups());
             TimeUnit.SECONDS.sleep(RANDOM_GENERATOR.nextInt(DELAY_RANGE) + MIN_DELAY_IN_SEC);
+
+            return Optional.of(modifiedData);
+
         } catch (JAXBException e) {
             e.printStackTrace();
             System.out.println("File not found or is not in valid xml format : " + cloudFile);
+
         } catch (InterruptedException e) {
             e.printStackTrace();
         }
-        return modifiedData;
+        return Optional.empty();
```

Comment:
I think this shouldn't be empty, but rather the newly modified data due to simulated corruption, which also means that `Optional` may not be suitable here.